dist/common/scripts/scylla_util.py

```diff
@@ -105,6 +109,141 @@ def curl(url, byte=False):
+
+    @staticmethod
+    def isNVME(gcpdiskobj):
+        """check if disk from GCP metadata is a NVME disk"""
+        if gcpdiskobj["interface"]=="NVME":
+            return True
+        return False
+
+    def __get_nvme_disks_from_metadata(self):
+        """get list of nvme disks from metadata server"""
+        import json
+        disksREST=self.__instance_metadata("disks")
+        disksobj=json.loads(disksREST)
+        nvmedisks=list(filter(self.isNVME, disksobj))
+        return nvmedisks
+
+    @property
+    def nvmeDiskCount(self):
+        """get # of nvme disks available for scylla raid"""
+        if self.__nvmeDiskCount is None:
+            try:
+                os_disks = self.__populate_os_disks()
+                ephemeral_disks = os_disks["ephemeral"]
+                count_os_disks=len(ephemeral_disks)
+            except Exception as e:
+                count_os_disks=0
+            nvme_metadata_disks = self.__get_nvme_disks_from_metadata()
+            count_metadata_nvme_disks=len(nvme_metadata_disks)
+            self.__nvmeDiskCount = count_os_disks if count_os_disks<count_metadata_nvme_disks else count_metadata_nvme_disks
+        return self.__nvmeDiskCount
+
+
+    @property
+    def instancetype(self):
+        """return the type of this instance"""
+        if self.__type is None:
+            self.__type = self.__instance_metadata("machine-type").split("/")[-1]
+        return self.__type
+
+    @property
+    def cpu(self):
+        """return the # of cpus of this instance"""
+        if self.__cpu is None:
+            self.__cpu = psutil.cpu_count()
+        return self.__cpu
+
+    @property
+    def memoryGB(self):
+        """return the size of memory in GB of this instance"""
+        if self.__memoryGB is None:
+            self.__memoryGB = psutil.virtual_memory().total/1024/1024/1024
+        return self.__memoryGB
+
+    def instance_size(self):
+        """Returns the size of the instance we are running in. i.e.: 2"""
+        return self.instancetype.split("-")[2]
+
+    def is_recommended_instance_size(self):
+        """if this instance has at least 2 cpus, it has a recommended size"""
+        if int(self.instance_size()) > 1:
+            return True
+        return False
+
+    # TODO below is in GB and is hardcoded, so this needs to be changed once the disks in GCP change
+    # note that they have 3TB physical devices actually, which they break into smaller 375GB disks and share the same mem with multiple machines
+    GCP_NVME_DISK_SIZE_2020=375
+
+    def is_recommended_instance(self):
+        if self.is_recommended_instance_size():
+            # at least 1:2GB cpu:ram ratio , GCP is at 1:4, so this should be fine
+            if self.cpu/self.memoryGB < 0.5:
+              # 30:1 Disk/RAM ratio must be kept at least(AWS), we relax this a little bit
+              # on GCP we are OK with 50:1 , n1-standard-2 can cope with 1 disk, not more
+              diskcount = self.nvmeDiskCount
```

Comment:
We're okay with 100:1

Reply:
the idea was to stay similar to AWS, should we relax it even more? (I am voting for no ;-)  stay at 50:1  )

Reply:
I don't understand.

Reply:
so I am voting for staying at current ratio
don't relax it more, I can change the comment and say scylla can handle more

Reply:
The 100:1 ratio is for Scylla, not the cloud.



AWS I3en have 70:1 ratio or so.